receiver/podmanreceiver/receiver.go

```diff
@@ -57,19 +69,90 @@ func newReceiver(
 		set:           set,
 	}
 
-	scrp, err := scraperhelper.NewScraper(typeStr, recv.scrape, scraperhelper.WithStart(recv.start))
+	return recv, err
+}
+
+func (r *receiver) registerMetricsConsumer(mc consumer.Metrics, set component.ReceiverCreateSettings) error {
+	r.metricsConsumer = mc
```

Comment:
Why need to keep `r.metricsConsumer`? Just to check that metrics are enabled? `r.metricsComponent` can be used instead right?